src/components/lessons/Git.md

```diff
@@ -45,6 +43,112 @@ git log
 
+Now you have created a new repo (short for repository) we will clone it on to our local machine. We first need the address for our repository which we can get by clicking on the new repo and on the green Code button. You should be able to copy the address from there. Once we have that we can do in to our text editor and open up our terminal. 
+
+## git clone
+
+To clone our repo we will type git clone (paste the address or your repo hero) and then press enter e.g :
+
+git clone git@github.com:Your-Name-Here/your-project.git
+
+This will create a copy of your repo . From here once we are in the folder You can go ahead and create and edit all you like. Once we are ready to save some changes we move on to git add.
+
+## git add
+
+If we type:
+
+“Git add .”
+
+It will tell git to track all of the files that are listed in the project. Alternatively we could just add the name of the file or folder we want it to track e.g:
+
+“git add index.html”
+
+Would tell it to add the index file.
+
+Once we have added files we then want to commit.
+
+## git commit
+
+Git commit is serious. It means we want to save our code here locally. Once we have added files in order to commit we will need to add a commit message. A commit message tells people what work you have done, its like a note. You can leave up to two messages here, the first of which is more like a heading and the second a more detailed explination. We add a message by using -m 'your message here'. So an example commit might look like:
+
+git commit -m 'fixed spelling mistakes ' -m 'this is the secondary message which is not required'
+
+## git push 
+
+ In order to make it live we must use another git command “git push” which means i want to push this live to a remote repository where my project is hosted. In order to push we need to have committed so once that is done we can go ahead. So in order to publish this properly we need to say :
+ git push origin main
+ 
+ Origin is a word that stands for the location of our repository and main is the branch we want to push to.
+
+
+## branching in git
+
+So here we are currently using the master/main which is the default branch in a repo. All of our contributions will go to that branch. We can also use another branch like if we wanted to add features to an app. So at the point of creation of the branch it is basically like a copy of the code but any changes made to either branch after that will not affect or be visible to one another . Any commits made will be made to that specific branch.
```

Comment:
This is just a note. GitHub is dropping the master branch and moving on to main branch. So using only the main branch in your lesson is fine. I think this will also make the readers less confused. 

Remove the space between "another" and period in this part "be visible to one another ."